File: src/ch04/code_4.py
from rich import print


def fibonacci(n: int) -> int:
    if n == 0:
        return 0
    elif n == 1:
        return 1
    return fibonacci(n-1) + fibonacci(n-2)


if __name__ == '__main__':
    result: int = fibonacci(n=10)
    print(result)
    

File: src/ch04/code_4.rs
fn fibonacci(n: i32) -> i32 {
    if n == 0 {
        return 0;
    } else if n == 1 {
        return 1;
    }
    fibonacci(n-1) + fibonacci(n-2)
}


fn main() {
    let result: i32 = fibonacci(10);
    println!("{}", result);
}